### PR TITLE
[adguard-home] fix  volumes subpath

### PR DESCRIPTION
**Pull request by abualy**
**Description of the change**

the goal of the change is to : 
- reflect subpath config in the values.yaml to show in the deployment

**Benefits**

- actually use subpath in volume mounts in the deployment when a user sets this parameter and expects it to work

**Possible drawbacks**

should be a seamless change for existing users, so can't think of any

**Applicable issues**

None

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [ ] (optional) Chart is using our [common](https://github.com/k8s-at-home/charts/tree/master/charts/common) library as a dependency.
- [ ] (optional) Variables are documented in the README.md

### CODE REVIEW

**Review by bjw-s — COMMENTED** [review]
Thanks a lot for taking the time to create this PR! 👍 
I have some reservations about the Service change, but since I'm not actually using the chart I will defer to the other maintainers. (I think @carpenike is actively running it?)

File: charts/stable/adguard-home/templates/service.yaml
Comment (abualy) [reply charts/stable/adguard-home/templates/service.yaml:29]: tbh, thinking about it, i agree, might as well keep 3000, so not to add to confusion, or match the port being used by the pods. but that could be in a separate PR
so for now i will change PR to handle only the `subPath` bug
thank you @bjw-s

File: charts/stable/adguard-home/templates/deployment.yaml
Comment (bjw-s) [inline charts/stable/adguard-home/templates/deployment.yaml:64]:
```suggestion
            {{- with .Values.persistence.work.subPath }}
              subPath: {{ . }}
            {{- end }}
```

File: charts/stable/adguard-home/templates/deployment.yaml
Comment (bjw-s) [inline charts/stable/adguard-home/templates/deployment.yaml:70]:
```suggestion
            {{- with .Values.persistence.conf.subPath }}
              subPath: {{ . }}
            {{- end }}
```

File: charts/stable/adguard-home/templates/deployment.yaml
Comment (bjw-s) [inline charts/stable/adguard-home/templates/deployment.yaml:90]:
```suggestion
            {{- with .Values.persistence.work.subPath }}
              subPath: {{ . }}
            {{- end }}
```

File: charts/stable/adguard-home/templates/deployment.yaml
Comment (bjw-s) [inline charts/stable/adguard-home/templates/deployment.yaml:96]:
```suggestion
            {{- with .Values.persistence.conf.subPath }}
              subPath: {{ . }}
            {{- end }}
```

**Review by bjw-s — CHANGES_REQUESTED** [review]
Looks like the CI is failing because `subPath` is optional. I added some suggestions that _should_ fix it.

**Comment by abualy** [comment]
thank you @bjw-s , i have very limited experience with helm templating, so this a good opportunity to learn :D

**Comment by abualy** [comment]
> Looks like the CI is failing because `subPath` is optional. I added some suggestions that _should_ fix it.

it seems weird, because i even tried setting supath parameter to default value:`""`, and not use conditional in the template, and still it gives the same linting error

File: charts/stable/adguard-home/templates/deployment.yaml
Comment (bjw-s) [inline charts/stable/adguard-home/templates/deployment.yaml:68]:
```suggestion
            {{- with .Values.persistence.config.subPath }}
```

File: charts/stable/adguard-home/templates/deployment.yaml
Comment (bjw-s) [inline charts/stable/adguard-home/templates/deployment.yaml:94]:
```suggestion
            {{- with .Values.persistence.config.subPath }}
```

**Review by bjw-s — CHANGES_REQUESTED** [review]
The item is called `persistence.config`, not `persistence.conf`, so it couldn't find that value when rendering the template. I missed that in my initial code suggestion, hopefully this fixes that 🙂

File: charts/stable/adguard-home/templates/deployment.yaml
Comment (abualy) [reply charts/stable/adguard-home/templates/deployment.yaml:68]: :facepalm:

**Review by bjw-s — APPROVED** [review]

**Pull request merged**
